### PR TITLE
Initial commit

### PR DESCRIPTION
**Pull request by yahavi**
New Maven plugin for generating dependency trees without compiling.
See description here: https://github.com/jfrog/maven-dep-tree/tree/dev#-maven-dependency-tree

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1 +1,86 @@
-# maven-dep-tree
\ No newline at end of file
+[![Scanned by Frogbot](https://raw.github.com/jfrog/frogbot/master/images/frogbot-badge.svg)](https://github.com/jfrog/frogbot#readme)
+[![Test](https://github.com/jfrog/maven-dep-tree/actions/workflows/test.yml/badge.svg?branch=main)](https://github.com/jfrog/maven-dep-tree/actions/workflows/test.yml)
+
+# 🪶 Maven Dependency Tree
+
+This Maven plugin reads the Maven dependencies of a given Maven project, and generates a dependency tree.
+This package was developed by JFrog, and is used by [Frogbot](https://github.com/jfrog/frogbot)
```
Comment (eyalbe4) [inline README.md:7]: [Frogbot] --> [JFrog Frogbot]

File: src/main/java/com/jfrog/Utils.java [hunk 2]
```diff
@@ -0,0 +1,121 @@
...
+                new DefaultProjectBuildingRequest(session.getProjectBuildingRequest());
+        buildingRequest.setProject(project);
+        DependencyNode root = dependencyGraphBuilder.buildDependencyGraph(buildingRequest, null);
+        Map<String, MavenDependencyNode> nodes = new HashMap<>();
+        populateDependencyMap(root, nodes);
+        return new MavenDepTreeResults(getGavString(root.getArtifact()), nodes);
+    }
+
+    /**
+     * Recursively populate the dependency nodes map from the given dependency tree.
```
Comment (eyalbe4) [inline src/main/java/com/jfrog/Utils.java:68]: populate --> populates

File: src/main/java/com/jfrog/Utils.java [hunk 3]
```diff
@@ -0,0 +1,121 @@
...
+    static void populateDependencyMap(DependencyNode node, Map<String, MavenDependencyNode> nodes) {
+        Artifact artifact = node.getArtifact();
+        MavenDependencyNode mavenDependencyNode = nodes.get(getGavString(artifact));
+        if (mavenDependencyNode == null) {
+            // Node does not exist in the Map - add it
+            mavenDependencyNode = new MavenDependencyNode(artifact.getScope());
+            nodes.put(getGavString(artifact), mavenDependencyNode);
+        } else {
+            // Node exists in the map - add the scope
+            mavenDependencyNode.addConfiguration(artifact.getScope());
```
Comment (eyalbe4) [inline src/main/java/com/jfrog/Utils.java:82]: Shouldn't this line be outside of the `else` clause? Meaning, shouldn't we add the configuration anyway, even if the node has just been added to the map?

**Review by eyalbe4 — CHANGES_REQUESTED** [review]
I think that project should be under the `com.jfrog.mavenDepTree` package, rather then under `com.jfrog`

File: src/main/java/com/jfrog/Utils.java (hunk 3 above)
Comment (yahavi) [reply src/main/java/com/jfrog/Utils.java:82]: 2 lines above it puts the configuration in the constructor

**Pull request merged**